### PR TITLE
Add view.detach() method to detach a view from its parent

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -4888,6 +4888,13 @@ def getView(self, iden=None, user=None):
 
         return view
 
+    def reqView(self, iden):
+        view = self.getView(iden)
+        if view is None: # pragma: no cover
+            mesg = f'No view with iden: {self.valu.get("iden")}'
+            raise s_exc.NoSuchView(mesg=mesg, iden=self.valu.get('iden'))
```
Comment (MichaelSquires) [inline synapse/cortex.py:4895]:
```suggestion
            mesg = f'No view with iden: {iden}'
            raise s_exc.NoSuchView(mesg=mesg, iden=iden)
```

I think `self.valu.get('iden')` is the cortex iden?

File: synapse/lib/stormtypes.py [hunk 2]
```diff
@@ -7080,6 +7086,15 @@ async def addNode(self, form, valu, props=None):
         else:
             await view.addNode(form, valu, props=props, user=self.runt.user)
 
+    async def detach(self):
+
+        view = self._reqView()
+        if not self.runt.isAdmin(gateiden=view.iden):
+            mesg = 'You must be an admin of the view to detach.'
+            raise s_exc.AuthDeny(mesg=mesg)
```
Comment (MichaelSquires) [inline synapse/lib/stormtypes.py:7094]:
```suggestion
            raise s_exc.AuthDeny(mesg=mesg, user=self.runt.user.iden, username=self.runt.user.name)
```
`user` and `username` are standard in the AuthDeny exception

File: synapse/tests/test_lib_stormtypes.py [hunk 3]
```diff
@@ -955,6 +955,28 @@ async def getseqn(genr, name, key):
...
+            await core.stormlist('$lib.view.get().detach()', opts={'view': iden})
+
+            with self.raises(s_exc.BadArg):
+                await core.nodes('$lib.view.get().detach()', opts={'view': iden})
+
+            view = core.reqView(iden)
+
+            self.none(view.parent)
+            self.none(view.info.get('parent'))
+
```
Comment (MichaelSquires) [inline synapse/tests/test_lib_stormtypes.py:979]: Do you want to try to re-parent `fork00` to make sure it fails? It should fail since `fork00` should have two layers (the default view layer and the new layer).

**Review by MichaelSquires — CHANGES_REQUESTED** [review]
Looks good! Some minor suggestions :)

File: synapse/cortex.py (hunk 1 above)
Comment (invisig0th) [reply synapse/cortex.py:4895]: it was a copy-pasta error from the View class in stormtypes. Good catch!

File: synapse/tests/test_lib_stormtypes.py (hunk 3 above)
Comment (invisig0th) [reply synapse/tests/test_lib_stormtypes.py:979]: that's handled in `synapse/tests/test_lib_view.py::ViewTest::test_view_set_parent` :+1:

**Review by MichaelSquires — APPROVED** [review]

File: synapse/lib/view.py [hunk 4]
```diff
@@ -144,6 +144,22 @@ async def __anit__(self, core, node):
         self.trigtask = None
         await self.initTrigTask()
 
+    @s_nexus.Pusher.onPushAuto('view:detach')
+    async def detach(self):
+        '''
+        Detach the view from it's parent but do not change the layers.
```
Comment (Cisphyx) [inline synapse/lib/view.py:150]:
```suggestion
        Detach the view from its parent but do not change the layers.
```

**Review by MichaelSquires — APPROVED** [review]

**Review by Cisphyx — APPROVED** [review]
